### PR TITLE
Add shop OpenAPI contracts and strengthen agent workflows

### PR DESCRIPTION
**Pull request by itemka**
## Summary

Add a generated OpenAPI 3.1 contract and contract testing for `shop-mvc-express`. Strengthen the repository's agent guidance for contract-first testing, external research, MCP security, skill validation, and subagent orchestration.

## Changes

[OpenAPI] - Generate the API contract from shared product Zod schemas and commit the generated artifact with drift and Redocly lint checks.

[Documentation routes] - Serve `/openapi.json` and Swagger UI under `/docs` outside production, including correct relative asset routing and existing Helmet protections.

[Contract tests] - Add Supertest coverage for product HTML routes, JSON and form submissions, validation boundaries, redirects, generated-contract parity, Swagger assets, error envelopes, and production route exclusion.

[CI and documentation] - Run workspace OpenAPI checks in CI and document the contract workflow in the app README, REST roadmap, and architecture model.

[Agent guidance] - Add contract-first testing rules, generated-file ownership, evidence-backed brainstorming, MCP prompt-injection boundaries, isolated skill-behavior validation, and dependency-aware subagent orchestration.

## Skill behavior validation

Ran 20 fresh isolated sessions: eight trigger cases, eight matched controls without the target repository skill, and four near misses.

- `brainstorm`: selected for durable order-processing and tenant-provisioning comparisons; rejected for an approved one-line configuration change. Added explicit source diversity, claim verification, dates, and recommendation confidence.
- `configuring-mcp`: selected for hostile third-party and shared documentation MCP configurations; rejected for an HTTP keep-alive explanation. Added untrusted-content boundaries and explicit human confirmation.
- `maintain-agent-docs`: selected for canonical skill/adapter maintenance and review; rejected for a general README edit. Added the exact trigger, control, near-miss, and evidence protocol missing from controls.
- `subagent-orchestration`: selected for backend dependency planning and multi-service authentication migration; rejected for a single-file typo. Improved topology choice, ownership, dependency gates, and reviewed-output reuse.

## Validation

- `pnpm --filter shop-mvc-express typecheck`
- `pnpm --filter shop-mvc-express test` — 15 tests passed
- `pnpm --filter shop-mvc-express build`
- `pnpm --filter shop-mvc-express openapi:check`
- ESLint and focused Prettier checks
- `pnpm run check:secrets`
- `pnpm run check:adapters`
- `git diff --cached --check`
- GitHub Actions was not run locally

## Risks

- The generated contract depends on Zod metadata and generator compatibility.
- Contract and Swagger routes are intentionally unavailable in production.
- This PR spans two review areas: application contracts and agent guidance.

## Rollback

Revert the two commits. No database migration or production API exposure is involved.

## Breaking changes

None.

### CODE REVIEW

File: docs/architecture/workspace.dsl [hunk 1]
```diff
@@ -21,7 +21,7 @@ workspace "nodejs-backend-mastery" "C4 model of the runnable apps in this monore
         }
 
         shopMvc = softwareSystem "shop-mvc-express" "Server-rendered product catalog: create/list products via HTML forms, Helmet security headers, Zod validation, in-memory product store." {
-            shopMvcApp = container "Express MVC app" "Routes + controllers render server-side HTML views; no separate frontend or API layer." "Node.js + Express"
+            shopMvcApp = container "Express MVC app" "Routes + controllers render server-side HTML views; no separate frontend. Publishes an OpenAPI 3.1 contract generated from the boundary Zod schemas at /openapi.json + Swagger UI at /docs (non-production only)." "Node.js + Express"
```
[the remark this answers is not in the document]
Comment (itemka) [reply docs/architecture/workspace.dsl:24]: @codex, it's fixed.
